src/github_utils.py

```diff
@@ -19,111 +20,104 @@ def get_pull_request_diff():
-            current_file = line.split(' b/')[-1].strip()
-            valid_lines[current_file] = set()
-            line_number = None  # Reset for new file
-            continue
-            
-        # Parse hunk header
-        if line.startswith('@@'):
+            current_file = line.split(' b/')[1].split()[0]
+            new_line = None
+        elif line.startswith('@@'):
             try:
-                # Extract new file line number (after +)
-                hunk_header = line.split('+', 1)[1].split(' ', 1)[0]
-                line_number = int(hunk_header.split(',')[0])
-                valid_lines[current_file].add(line_number)
+                # Extract new file range from hunk header
+                new_range = line.split('+')[1].split()[0].split(',')
+                new_start = int(new_range[0])
+                new_line_count = int(new_range[1]) if len(new_range) > 1 else 1
+                new_line = new_start
+                hunks.append({
+                    'file': current_file,
+                    'start': new_start,
+                    'end': new_start + new_line_count - 1,
+                    'lines': set(range(new_start, new_start + new_line_count))
+                })
             except (IndexError, ValueError):
-                line_number = None  # Invalid hunk format
-            continue
-            
-        # Only process lines if we have valid line_number
-        if line_number is None:
-            continue
-            
-        # Track added lines
-        if line.startswith('+'):
-            valid_lines[current_file].add(line_number)
-            line_number += 1
-        # Track context lines (non-diff lines)
-        elif line.startswith(' '):
-            line_number += 1
-            
-    return valid_lines
-
+                new_line = None
+        elif new_line is not None:
+            if line.startswith('+'):
+                new_line += 1
+            elif line.startswith(' '):
+                new_line += 1
+                
+    return hunks
+
+def validate_comment(comment: dict, hunks: List[dict]) -> Optional[dict]:
+    """Validate comment against actual diff hunks"""
+    for hunk in hunks:
+        if (hunk['file'] == comment['path'] and
+            comment['start_line'] >= hunk['start'] and
+            comment['end_line'] <= hunk['end']):
+            return {
+                'path': comment['path'],
+                'start_line': comment['start_line'],
+                'line': comment['end_line'],
+                'body': comment['body'],
+                'start_side': 'RIGHT',
+                'side': 'RIGHT'
+            }
+    print(f"Invalid range: {comment['path']} {comment['start_line']}-{comment['end_line']}")
+    return None
 
 
 def post_review_comment(comments, diff_content):
-    """Post validated review comments to GitHub"""
     g = Github(os.getenv('GIT_TOKEN'))
-
-    # Load GitHub event data
     event_path = os.getenv('GITHUB_EVENT_PATH')
+    
```

Comment:
Final validation step checks that the start_line is before the end_line and the file exists in the extracted hunks.
```suggestion
|
```
N/A